### ISSUE TITLE
Question regarding dataset information

### ISSUE THREAD

**Issue by maciejdomagala** [jdb78/pytorch-forecasting#191]
Hi, I had a question regarding information passed via model instance.

Scenario:
I am creating a training TimeSeriesDataSet object and validation TimeSeriesDataSet by using 'from_dataset' method and disabling randomization.
Next, I am initializing the TFT model by using 'from_dataset' functionality. After training I am saving the model using ModelCheckpoint callback.
I would later like to use my model to create a prediction on new data, somewhere else in my code architecture.

Question:
To avoid some statistical inconsistency and leakage I would like to use the same encoding/scaling/normalizing that was used for the training data. Is the information about EncoderNormalizer/categorical features encoding etc. passed via model instance?  I noticed that I we can pass target normalization by

```
if "output_transformer" not in kwargs:
            kwargs["output_transformer"] = dataset.target_normalizer
```
but categorical features encoding is not passed this way. Should I also pass the training TimeSeriesDataSet instance for the model caching to reuse it while predicting or is there other way?

**Comment by jdb78** [comment]
You can go two ways here.

1. Just pass the pandas dataframe to the model's `.predict()` method. The model saves the dataset parameters (attribute `dataset_parameters`) when you instantiate it from a dataset and will use them to create a dataset from the dataframe which is turned into a dataloader and passed through the network.

2. If you want fine-grained control, serialise the dataset parameters to disc (get them with `dataset.get_parameters()`) and create new TimeSeriesDataSet instances by using the `.from_parameters()` method.

**Comment by maciejdomagala** [comment]
Alright, I believe that I missed the fact that we can apply dataframe for prediction without the need to create a dataloader (for which dataset would be necessary). Great news, thanks a lot!
